components/LocationInput.tsx

```diff
@@ -8,126 +8,164 @@ import React, {
+        geocoder.setBbox([0, 0, 0, 0]);
       }
 
+      geocoderContainer.current?.appendChild(geocoder.onAdd(map.current));
+
       const marker = new mapboxgl.Marker({
         draggable: false,
         color: '#FFD053',
-      }).setLngLat([lng, lat]);
+      }).setLngLat([originalLat, originalLng]);
 
-      map.current.on('load', () => setIsMapReady(true));
+      map.current.on('load', () => {
+        if (shouldZoomIntoOriginal) {
+          geocoder.query(`${originalLat}, ${originalLng}`).setFlyTo(canEdit);
+        }
+      });
 
       map.current.on('idle', () => {
         // Enables fly to animation on search
         geocoder.setFlyTo(true);
       });
 
-      map.current.on('movestart', () => {
-        setIsSubmissionEnabled(true);
-      });
-
-      let savedLat = originalLng;
-      let savedLng = originalLat;
+      let savedLat = originalLat;
+      let savedLng = originalLng;
 
       const moveHandler = () => {
         const newLng = map.current!.getCenter().lng;
         const newLat = map.current!.getCenter().lat;
 
-        setLng(newLng);
-        setLat(newLat);
-        setLngExternal(newLng);
-        setLatExternal(newLat);
+        setMapCoordinates({ longitude: newLng, latitude: newLat });
 
         savedLat = newLat;
         savedLng = newLng;
+        updateMarker(
+          marker,
+          map.current!,
+          zoomLevelThreshold,
+          newLng,
+          newLat,
+          canEdit
+        );
 
         setZoom(map.current!.getZoom());
-        updateMarker(marker, map.current!, zoomLevelThreshold, newLng, newLat);
       };
 
-      map.current.on('move', moveHandler);
+      map.current.on('movestart', () => {
+        if (popup) {
+          popup.remove();
+        }
+      });
 
       map.current.on('moveend', () => {
-        const isPastZoomThreshold = map.current!.getZoom() > zoomLevelThreshold;
+        const isPastZoomThreshold =
+          map.current!.getZoom() >= zoomLevelThreshold;
         if (isPastZoomThreshold) {
+          moveHandler();
+
           // update the search box location based on the final latitude/longitude
           geocoder.query(`${savedLat}, ${savedLng}`).setFlyTo(false);
+
+          if (popup === null && map.current && canEdit) {
+            popup = new mapboxgl.Popup({
+              offset: [0, 10],
+              anchor: 'top',
+              closeOnClick: false,
+              closeButton: false,
+              className: 'site-map',
+            })
+              .setLngLat([savedLng, savedLat])
+              .setHTML('<p>Drag map to pinpoint exact location.</p>')
+              .addTo(map.current);
+          }
         }
 
         return setIsSubmissionEnabled(isPastZoomThreshold);
       });
     }
   }, [
-    lat,
-    lng,
+    canEdit,
+    shouldZoomIntoOriginal,
+    originalLat,
+    originalLng,
     setIsSubmissionEnabled,
-    setLatExternal,
-    setLngExternal,
+    setMapCoordinates,
     zoom,
     zoomLevelThreshold,
   ]);
```

Comment:
A future issue: I think we should remove most if not all of these dependencies. The map ref should strictly interacted with via its set/get methods since recreating it might be causing issues... not for this PR.

Reply:
E.x. I'm noticing dragging the map doesn't exactly drag the marker the same way anymore... not too big a deal for now since it works well with lat/lng